### PR TITLE
Estate Info Story Runner

### PR DESCRIPTION
**Pull request by eccles**
Problem:
The estate_info sample code is missing from the
story runner.

Solution:
Added a Composite endpoint designed to accommodate
logic spanning multiple endpoints. Added this code
to story runner and new functional test
test_runner_estate_info.

Signed-off-by: Paul Hewlett <phewlett76@gmail.com>

### CODE REVIEW

File: functests/test_resources/door_entry_story.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ steps:
       description: Create a door in Paris at Jitsuin offices.
     signature:
       attributes:
-        arc_display_name: Jitsuin Paris Front Door
+        arc_display_name: Jitsuin Paris Front Door Demo5
```
Comment (JAG-UK) [inline functests/test_resources/door_entry_story.yaml:17]: Probably didn't want to commit this Demo5 thing. Maybe in general we should be encouraging arc_namespace for this anyway (?)

File: examples/scan_test.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
+"""
```
Comment (JAG-UK) [inline examples/scan_test.py:1]: This is cool but does it belong in this PR?

File: archivist/composite.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    def __str__(self) -> str:
+        return f"CompositeClient({self._archivist.url})"
+
+    def estate_info(self):
+        """
+        Evaluate health of the various assets and events in the sytem
+
+        The report is emitted using LOGGER.info statements
+        """
+        # for around 550 events and 250 assets this can take about a 90s...
```
Comment (JAG-UK) [inline archivist/composite.py:59]: Does it not use the quick counting method (the COUNT header)?

File: archivist/composite.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+# pylint:disable=unused-import      # To prevent cyclical import errors forward referencing is used
+# pylint:disable=cyclic-import      # but pylint doesn't understand this feature
+from . import archivist as type_helper
+
+
+LOGGER = getLogger(__name__)
+
+
+class _CompositeClient:
```
Comment (JAG-UK) [inline archivist/composite.py:35]: I don't think we really need this. The double-check method and duplicate finder were there from the very early days when it was possible to pick up double records from blockchain caches and things. That doesn't happen anymore, so the YAML that just counts the 3 things is enough.

File: functests/test_resources/estate_info_story.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      action: EVENTS_COUNT
+      description: Count all events
+      print_response: true
+
+  - step:
+      action: LOCATIONS_COUNT
+      description: Count all locations
+      print_response: true
+
+  - step:
```
Comment (JAG-UK) [inline functests/test_resources/estate_info_story.yaml:25]: I don't think we really need this. The double-check method and duplicate finder were there from the very early days when it was possible to pick up double records from blockchain caches and things. That doesn't happen anymore, so I think just counting the 3 things is enough. The question comes though, how do we pretty-print the answers/count numbers?

**Review by JAG-UK — CHANGES_REQUESTED** [review]

File: examples/scan_test.py (hunk 2 above)
Comment (eccles) [reply examples/scan_test.py:1]: not really - its a drive-by - would you like it elsewhere?
I am waiting for the actual code to come back from JS when he gets it done...

File: functests/test_resources/door_entry_story.yaml (hunk 1 above)
Comment (eccles) [reply functests/test_resources/door_entry_story.yaml:17]: probably best solved by sourceing arc_namespace from the environment - just haven't got to that part yet...

File: archivist/composite.py (hunk 3 above)
Comment (eccles) [reply archivist/composite.py:59]: yes it does for the simple case - the cross-checking code requires the list() calls but if you think this is no longer needed than can be simplified greatly...

**Review by JAG-UK — APPROVED** [review]

**Review by JAG-UK — APPROVED** [review]

**Pull request merged**